packages/drivekit_driver_data/flutter_drivekit_driver_data_android/lib/src/model_adapter.dart

```diff
@@ -0,0 +1,570 @@
+          .whereNotNull()
+          .map((e) => e.toModelImplementation())
+          .toList(),
+    );
+  }
+}
+
+/// Adapts the [PigeonAdvancedFuelEstimation] class to the corresponding class
+/// in the model.
+extension PigeonAdvancedFuelEstimationAdapter on PigeonAdvancedFuelEstimation {
+  /// Converts a [PigeonAdvancedFuelEstimation] to a corresponding model class.
+  AdvancedFuelEstimation toModelImplementation() {
+    return AdvancedFuelEstimation(
+      fuelEstimationContext: fuelEstimationContext
+          .whereNotNull()
+          .map((e) => e.toModelImplementation())
+          .toList(),
+    );
+  }
+}
+
+/// Adapts the [PigeonTrip] class to the corresponding class
+/// in the model.
+extension PigeonTripAdapter on PigeonTrip {
+  /// Converts a [PigeonTrip] to a corresponding model class.
+  Trip toModelImplementation() {
+    return Trip(
+      itinId: itinId,
+      comments: comments
+          .whereNotNull()
+          .map((e) => e.toModelImplementation())
+          .toList(),
+      tripStatistics: tripStatistics?.toModelImplementation(),
+      ecoDriving: ecoDriving?.toModelImplementation(),
+      fuelEstimation: fuelEstimation?.toModelImplementation(),
+      safety: safety?.toModelImplementation(),
+      advancedEcoDriving: advancedEcoDriving?.toModelImplementation(),
+      advancedFuelEstimation: advancedFuelEstimation?.toModelImplementation(),
+      advancedSafety: advancedSafety?.toModelImplementation(),
+      pollutants: pollutants?.toModelImplementation(),
+      tireWear: tireWear?.toModelImplementation(),
+      brakeWear: brakeWear?.toModelImplementation(),
+      driverDistraction: driverDistraction?.toModelImplementation(),
+      itineraryData: itineraryData?.toModelImplementation(),
+      endDate: endDate,
+      logbook: logbook?.toModelImplementation(),
+      safetyEvents: safetyEvents
+          ?.whereNotNull()
+          .map((e) => e.toModelImplementation())
+          .toList(),
+      speedingStatistics: speedingStatistics?.toModelImplementation(),
+      energyEstimation: energyEstimation?.toModelImplementation(),
+      advancedEnergyEstimation: advancedEnergyEstimation
+          ?.whereNotNull()
+          .map((e) => e.toModelImplementation())
+          .toList(),
+      startDate: startDate,
+      departureCity: departureAddress,
+      arrivalCity: arrivalAddress,
+      departureAddress: departureAddress,
+      arrivalAddress: arrivalAddress,
+      vehicleId: vehicleId,
+      metaData: convertMetadata(metaData),
+      transportationMode: transportationMode,
+      unscored: unscored,
+      calls:
+          calls?.whereNotNull().map((e) => e.toModelImplementation()).toList(),
+      speedLimitContexts: speedLimitContexts
+          ?.whereNotNull()
+          .map((PigeonSpeedLimitContext e) => e.toModelImplementation())
+          .toList(),
+    );
+  }
+}
+
+/// Convert a the metadata map from pigeon to a map of non-nullable strings.
```

Comment:
"Convert ~a~ the"

Reply:
bien vu, corrigé sur 168fba8e9a76cb16947529e299e4b0a367753157